extensions-core/kafka-indexing-service/src/main/java/org/apache/druid/data/input/kafkainput/KafkaInputFormat.java

```diff
@@ -0,0 +1,167 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.data.input.kafkainput;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.base.Preconditions;
+import org.apache.druid.data.input.InputEntity;
+import org.apache.druid.data.input.InputEntityReader;
+import org.apache.druid.data.input.InputFormat;
+import org.apache.druid.data.input.InputRowSchema;
+import org.apache.druid.data.input.impl.ByteEntity;
+import org.apache.druid.data.input.impl.TimestampSpec;
+import org.apache.druid.data.input.kafka.KafkaRecordEntity;
+import org.apache.druid.java.util.common.DateTimes;
+
+import javax.annotation.Nullable;
+import java.io.File;
+import java.util.Objects;
+
+public class KafkaInputFormat implements InputFormat
+{
+  private static final String DEFAULT_HEADER_LABEL_PREFIX = "kafka.header.";
+  private static final String DEFAULT_TIMESTAMP_LABEL_PREFIX = "kafka.";
+  private static final String DEFAULT_KEY_LABEL_PREFIX = "kafka.";
+
+  private final KafkaHeaderFormat headerFormat;
+  private final InputFormat valueFormat;
+  private final InputFormat keyFormat;
+  private final String headerLabelPrefix;
+  private final String keyLabelPrefix;
+  private final String recordTimestampLabelPrefix;
+
+  public KafkaInputFormat(
+      @JsonProperty("headerFormat") @Nullable KafkaHeaderFormat headerFormat,
+      @JsonProperty("keyFormat") @Nullable InputFormat keyFormat,
+      @JsonProperty("valueFormat") InputFormat valueFormat,
+      @JsonProperty("headerLabelPrefix") @Nullable String headerLabelPrefix,
+      @JsonProperty("keyLabelPrefix") @Nullable String keyLabelPrefix,
+      @JsonProperty("recordTimestampLabelPrefix") @Nullable String recordTimestampLabelPrefix
+  )
+  {
+    this.headerFormat = headerFormat;
+    this.keyFormat = keyFormat;
+    this.valueFormat = Preconditions.checkNotNull(valueFormat, "valueFormat");
```

Comment:
can we provide a more meaningful error message here, e.g.
```suggestion
    this.valueFormat = Preconditions.checkNotNull(valueFormat, "valueFormat must not be null");
```